src/core/system.js

```diff
@@ -310,19 +311,21 @@ export default class Store {
 
 }
 
-function combinePlugins(plugins, toolbox) {
+function combinePlugins(plugins, toolbox, pluginLoadType) {
   if(isObject(plugins) && !isArray(plugins)) {
     return assignDeep({}, plugins)
   }
 
   if(isFunc(plugins)) {
-    return combinePlugins(plugins(toolbox), toolbox)
+    return combinePlugins(plugins(toolbox), toolbox, pluginLoadType)
   }
 
   if(isArray(plugins)) {
+    const dest = pluginLoadType === "chain" ? toolbox.getComponents() : {}
```

Comment:
revised comment... and then check for `pluginOptions.pluginLoadType` here

Reply:
Ok, It may open some future tweaking this way